sub/sd_ass.c

```diff
@@ -288,46 +288,53 @@ static int init(struct sd *sd)
     return 0;
 }
 
+static inline bool is_escaped(const char *str, size_t pos)
+{
+    size_t count = 0;
+    while (pos > 0 && str[--pos] == '\\')
+        ++count;
+    return count % 2;
+}
+
 // Check if subtitle has events that would cause it to be animated inside {}
-static bool is_animated(char *s)
+static bool is_animated(const char *str)
 {
-    bool in_tag = false;
-    bool valid_event = false;
-    bool valid_tag = false;
-    while (*s) {
-        if (!in_tag && s[0] == '{')
-            in_tag = true;
-        if (s[0] == '\\') {
-            s++;
-            if (!s[0])
+    const char *begin = str;
+    while ((begin = strchr(begin, '{'))) {
+        if (!begin)
+            return false;
+
+        if (is_escaped(str, begin - str)) {
+            ++begin;
+            continue;
+        }
+
+        const char *end = begin;
+        while ((end = strchr(end, '}'))) {
+            if (!is_escaped(str, end - str))
                 break;
-            if (s[0] == 'k' || s[0] == 'K' || s[0] == 't') {
-                valid_event = true;
-                continue;
-            // just bruteforce the multi-letter ones
-            } else if (s[0] == 'f') {
-                if (!strncmp(s, "fad", 3)) {
-                    valid_event = true;
-                    continue;
-                }
-            } else if (s[0] == 'm') {
-                if (!strncmp(s, "move", 4)) {
-                    valid_event = true;
-                    continue;
+            ++end;
+        }
+        if (!end)
+            return false;
+
+        const char *pos = begin + 1;
+        while ((pos = memchr(pos, '\\', end - pos))) {
+            if (!is_escaped(str, pos - str)) {
```

Comment:
Tags cannot be escaped. However, spaces and tabs between the backslash and the tag name are ignored, and should be skipped here. I'd recommend:
```
while (*pos == '\\')
  pos++;
while (*pos == ' ' || *pos == '\t')
  pos++;
```

And then start the checks from `pos[0]` instead of `pos[1]`, and remove the `++pos` at the end of the loop.